atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -41,25 +41,95 @@
  */
 public interface Transaction {
 
+    /**
+     * Returns a mapping of rows to {@link RowResult}s within {@code tableRef} for the specified {@code rows}, loading
+     * columns according to the provided {@link ColumnSelection}.
+     *
+     * The returned {@link SortedMap} is sorted on the byte order of row keys; the ordering of the input parameter
+     * {@code rows} is irrelevant. Access to the returned {@link SortedMap} is permitted both with the original byte
+     * array, as well as with byte arrays that are equal in terms of {@link java.util.Arrays#equals(byte[], byte[])}.
+     *
+     * If there are rows with no cells matching the provided {@link ColumnSelection}, they will not be present in the
+     * {@link Map#keySet()} of the output map at all.
+     *
+     * @param tableRef table to load rows from
+     * @param rows rows to be loaded
+     * @param columnSelection columns to load from the given rows
+     * @return a mapping of rows to the columns matching the provided column selection
+     */
     @Idempotent
     SortedMap<byte[], RowResult<byte[]>> getRows(
             TableReference tableRef,
             Iterable<byte[]> rows,
             ColumnSelection columnSelection);
 
+    /**
+     * Returns a mapping of rows to {@link BatchingVisitable}s (which may be thought of as iterators over columns)
+     * within {@code tableRef} for the specified {@code rows}, where the columns fall within the provided
+     * {@link BatchColumnRangeSelection}. The single provided {@link BatchColumnRangeSelection} applies to all of the
+     * rows.
+     *
+     * The returned {@link BatchingVisitable}s are guaranteed to return cells matching the predicate, sorted on
+     * (ascending) byte ordering.
+     *
+     * It is guaranteed that the {@link Map#keySet()} of the returned map has the same elements as {@code rows} by
+     * identity, even if there are rows where no columns match the predicate. Random access for the returned map
+     * should ONLY be performed through the original byte arrays that were passed in.
```

Comment:
Can we tighten this up? It'd be easy to (under the hood) just wrap with an ImmutableSortedMap and so have a consistent contract between this and getRows.

Reply:
Like, arguably we should have a consistent contract between all our methods.

Reply:
Yep, changed this (and added appropriate tests).